lib/helpers/utils.js

```diff
@@ -7374,6 +7380,63 @@ function fileListToComponents(fileList) {
   return components;
 }
 
+/**
+ * Parse dependency from the `[package]` node in cargo.toml or cargo.lock
+ * @param {Object} packageNode
+ * @returns {Object} dependency info
+ * @throws {Error} if dependency name or version is invalid
+ */
+function parseCargoDependencyFromPackageNode(packageNode) {
+  const pkg = {};
+  const pkgName = packageNode["name"];
+  let group = dirname(pkgName.toString());
+  if (group === ".") {
+    group = "";
+  }
+  const pkgChecksum = packageNode["checksum"];
+  const pkgVersion = packageNode["version"];
+  const pkgAuthors = packageNode["authors"];
+  const pkgHomepage = packageNode["homepage"];
+  const pkgRepository = packageNode["repository"];
+  const pkgLicense = packageNode["license"];
+  const pkgDependencies = packageNode["dependencies"];
+
+  if (!pkgName || !pkgVersion) {
+    throw new Error("name or version is not defined");
+  }
+
+  if (pkgChecksum) {
+    pkg._integrity = `${pkgChecksum}`;
+  }
+  if (pkgName) {
+    pkg.group = group;
+    pkg.name = basename(pkgName.toString());
+  }
+  if (pkgVersion) {
+    pkg.version = pkgVersion;
+  }
+  if (pkgAuthors) {
+    if (Object.prototype.toString.call(pkgAuthors) === "[object Array]") {
+      pkg.author = pkgAuthors.join(",");
+    } else {
+      pkg.author = Object.prototype.toString.call(pkgAuthors);
+    }
+  }
+  if (pkgHomepage) {
+    pkg.homepage = { url: pkgHomepage };
+  }
+  if (pkgRepository) {
+    pkg.repository = { url: pkgRepository };
+  }
+  if (pkgLicense) {
+    pkg.license = pkgLicense;
```

Comment:
Repotests is [complaining](https://github.com/CycloneDX/cdxgen/actions/runs/12947327972/job/36116464090?pr=1595#step:98:23) about this line. Any ideas?